// wallet/model/inlineObject50.ts
/**
 * wallet-api
 * API
 *
 * The version of the OpenAPI document: 1.1.6
 * Contact: development@wallet.inc
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { RequestFile } from './models';

export class InlineObject50 {
    'metaValue': string;
    'transactionNumber': string;
    'checkAmount': number;

    static discriminator: string | undefined = undefined;

    static attributeTypeMap: Array<{name: string, baseName: string, type: string}> = [
        {
            "name": "metaValue",
            "baseName": "metaValue",
            "type": "string"
        },
        {
            "name": "transactionNumber",
            "baseName": "transactionNumber",
            "type": "string"
        },
        {
            "name": "checkAmount",
            "baseName": "checkAmount",
            "type": "number"
        }    ];

    static getAttributeTypeMap() {
        return InlineObject50.attributeTypeMap;
    }
}
