// wallet/model/inlineResponse2006.ts
/**
 * wallet-api
 * API
 *
 * The version of the OpenAPI document: 1.1.6
 * Contact: development@wallet.inc
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { RequestFile } from './models';

export class InlineResponse2006 {
    'walletConfiguration': string;
    'digitalWalletSession': string;
    'staticVoucher': string;
    'staticVoucherCampaign': string;
    'dynamicVoucher': string;
    'advertisementCreditScan': string;
    'advertisementCredit': string;
    'performance': string;
    'newsArticle': string;
    'merchantCredit': string;
    'member': string;
    'membershipTier': string;

    static discriminator: string | undefined = undefined;

    static attributeTypeMap: Array<{name: string, baseName: string, type: string}> = [
        {
            "name": "walletConfiguration",
            "baseName": "WalletConfiguration",
            "type": "string"
        },
        {
            "name": "digitalWalletSession",
            "baseName": "DigitalWalletSession",
            "type": "string"
        },
        {
            "name": "staticVoucher",
            "baseName": "StaticVoucher",
            "type": "string"
        },
        {
            "name": "staticVoucherCampaign",
            "baseName": "StaticVoucherCampaign",
            "type": "string"
        },
        {
            "name": "dynamicVoucher",
            "baseName": "DynamicVoucher",
            "type": "string"
        },
        {
            "name": "advertisementCreditScan",
            "baseName": "AdvertisementCreditScan",
            "type": "string"
        },
        {
            "name": "advertisementCredit",
            "baseName": "AdvertisementCredit",
            "type": "string"
        },
        {
            "name": "performance",
            "baseName": "Performance",
            "type": "string"
        },
        {
            "name": "newsArticle",
            "baseName": "NewsArticle",
            "type": "string"
        },
        {
            "name": "merchantCredit",
            "baseName": "MerchantCredit",
            "type": "string"
        },
        {
            "name": "member",
            "baseName": "Member",
            "type": "string"
        },
        {
            "name": "membershipTier",
            "baseName": "MembershipTier",
            "type": "string"
        }    ];

    static getAttributeTypeMap() {
        return InlineResponse2006.attributeTypeMap;
    }
}
