// wallet/model/inlineObject55.ts
/**
 * wallet-api
 * API
 *
 * The version of the OpenAPI document: 1.1.6
 * Contact: development@wallet.inc
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { RequestFile } from './models';

export class InlineObject55 {
    'isOver21Required': boolean;
    'optOutConfirmedCustomerReceives': string;
    'optOutConfirmedResponse': string;
    'optOutKeyword': string;
    'optInConfirmedCustomerReceives': string;
    'optInConfirmedResponse': string;
    'optInKeyword': string;
    'estimatedMessagesPerMonth': number;
    'listName': string;
    'isActive': boolean;
    'phoneNumberID': string;

    static discriminator: string | undefined = undefined;

    static attributeTypeMap: Array<{name: string, baseName: string, type: string}> = [
        {
            "name": "isOver21Required",
            "baseName": "isOver21Required",
            "type": "boolean"
        },
        {
            "name": "optOutConfirmedCustomerReceives",
            "baseName": "optOutConfirmedCustomerReceives",
            "type": "string"
        },
        {
            "name": "optOutConfirmedResponse",
            "baseName": "optOutConfirmedResponse",
            "type": "string"
        },
        {
            "name": "optOutKeyword",
            "baseName": "optOutKeyword",
            "type": "string"
        },
        {
            "name": "optInConfirmedCustomerReceives",
            "baseName": "optInConfirmedCustomerReceives",
            "type": "string"
        },
        {
            "name": "optInConfirmedResponse",
            "baseName": "optInConfirmedResponse",
            "type": "string"
        },
        {
            "name": "optInKeyword",
            "baseName": "optInKeyword",
            "type": "string"
        },
        {
            "name": "estimatedMessagesPerMonth",
            "baseName": "estimatedMessagesPerMonth",
            "type": "number"
        },
        {
            "name": "listName",
            "baseName": "listName",
            "type": "string"
        },
        {
            "name": "isActive",
            "baseName": "isActive",
            "type": "boolean"
        },
        {
            "name": "phoneNumberID",
            "baseName": "phoneNumberID",
            "type": "string"
        }    ];

    static getAttributeTypeMap() {
        return InlineObject55.attributeTypeMap;
    }
}
