// wallet/model/inlineObject53.ts
/**
 * wallet-api
 * API
 *
 * The version of the OpenAPI document: 1.1.6
 * Contact: development@wallet.inc
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { RequestFile } from './models';

export class InlineObject53 {
    'watsonConversationWorkplaceID'?: string;
    'watsonPassword'?: string;
    'watsonUsername'?: string;
    'isConnectedToWatson'?: boolean;
    'helpDeskQueueResponse': string;
    'helpDeskKeyword': string;
    'helpResponse': string;
    'stopResponse': string;
    'messageFooter': string;
    'termsOfServiceURL': string;
    'privacyPolicyURL': string;
    'companyName': string;

    static discriminator: string | undefined = undefined;

    static attributeTypeMap: Array<{name: string, baseName: string, type: string}> = [
        {
            "name": "watsonConversationWorkplaceID",
            "baseName": "watsonConversationWorkplaceID",
            "type": "string"
        },
        {
            "name": "watsonPassword",
            "baseName": "watsonPassword",
            "type": "string"
        },
        {
            "name": "watsonUsername",
            "baseName": "watsonUsername",
            "type": "string"
        },
        {
            "name": "isConnectedToWatson",
            "baseName": "isConnectedToWatson",
            "type": "boolean"
        },
        {
            "name": "helpDeskQueueResponse",
            "baseName": "helpDeskQueueResponse",
            "type": "string"
        },
        {
            "name": "helpDeskKeyword",
            "baseName": "helpDeskKeyword",
            "type": "string"
        },
        {
            "name": "helpResponse",
            "baseName": "helpResponse",
            "type": "string"
        },
        {
            "name": "stopResponse",
            "baseName": "stopResponse",
            "type": "string"
        },
        {
            "name": "messageFooter",
            "baseName": "messageFooter",
            "type": "string"
        },
        {
            "name": "termsOfServiceURL",
            "baseName": "termsOfServiceURL",
            "type": "string"
        },
        {
            "name": "privacyPolicyURL",
            "baseName": "privacyPolicyURL",
            "type": "string"
        },
        {
            "name": "companyName",
            "baseName": "companyName",
            "type": "string"
        }    ];

    static getAttributeTypeMap() {
        return InlineObject53.attributeTypeMap;
    }
}
