// wallet/model/inlineObject29.ts
/**
 * wallet-api
 * API
 *
 * The version of the OpenAPI document: 1.1.6
 * Contact: development@wallet.inc
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { RequestFile } from './models';

export class InlineObject29 {
    'selectedRegisters'?: Array<string>;
    'endDateTime': Date;
    'startDateTime': Date;
    'routingIDs'?: Array<string>;

    static discriminator: string | undefined = undefined;

    static attributeTypeMap: Array<{name: string, baseName: string, type: string}> = [
        {
            "name": "selectedRegisters",
            "baseName": "selectedRegisters",
            "type": "Array<string>"
        },
        {
            "name": "endDateTime",
            "baseName": "endDateTime",
            "type": "Date"
        },
        {
            "name": "startDateTime",
            "baseName": "startDateTime",
            "type": "Date"
        },
        {
            "name": "routingIDs",
            "baseName": "routingIDs",
            "type": "Array<string>"
        }    ];

    static getAttributeTypeMap() {
        return InlineObject29.attributeTypeMap;
    }
}
