// wallet/model/inlineObject69.ts
/**
 * wallet-api
 * API
 *
 * The version of the OpenAPI document: 1.1.6
 * Contact: development@wallet.inc
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { RequestFile } from './models';

export class InlineObject69 {
    'ein'?: string;
    'phoneNumber': string;
    'country': string;
    'zip': string;
    'state': string;
    'city': string;
    'streetAddress2': string;
    'streetAddress1': string;
    'merchantType': string;
    'companyName': string;
    'hearAboutUsDetails': string;
    'hearAboutUs': string;
    'password': string;
    'email': string;
    'lastName': string;
    'firstName': string;

    static discriminator: string | undefined = undefined;

    static attributeTypeMap: Array<{name: string, baseName: string, type: string}> = [
        {
            "name": "ein",
            "baseName": "ein",
            "type": "string"
        },
        {
            "name": "phoneNumber",
            "baseName": "phoneNumber",
            "type": "string"
        },
        {
            "name": "country",
            "baseName": "country",
            "type": "string"
        },
        {
            "name": "zip",
            "baseName": "zip",
            "type": "string"
        },
        {
            "name": "state",
            "baseName": "state",
            "type": "string"
        },
        {
            "name": "city",
            "baseName": "city",
            "type": "string"
        },
        {
            "name": "streetAddress2",
            "baseName": "streetAddress2",
            "type": "string"
        },
        {
            "name": "streetAddress1",
            "baseName": "streetAddress1",
            "type": "string"
        },
        {
            "name": "merchantType",
            "baseName": "merchantType",
            "type": "string"
        },
        {
            "name": "companyName",
            "baseName": "companyName",
            "type": "string"
        },
        {
            "name": "hearAboutUsDetails",
            "baseName": "hearAboutUsDetails",
            "type": "string"
        },
        {
            "name": "hearAboutUs",
            "baseName": "hearAboutUs",
            "type": "string"
        },
        {
            "name": "password",
            "baseName": "password",
            "type": "string"
        },
        {
            "name": "email",
            "baseName": "email",
            "type": "string"
        },
        {
            "name": "lastName",
            "baseName": "lastName",
            "type": "string"
        },
        {
            "name": "firstName",
            "baseName": "firstName",
            "type": "string"
        }    ];

    static getAttributeTypeMap() {
        return InlineObject69.attributeTypeMap;
    }
}
