// wallet/model/inlineObject51.ts
/**
 * wallet-api
 * API
 *
 * The version of the OpenAPI document: 1.1.6
 * Contact: development@wallet.inc
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { RequestFile } from './models';

export class InlineObject51 {
    'isShortCodeTimelineUnderstood': boolean;
    'isPricingUnderstood': boolean;
    'isShortCodeTransferUnderstood': boolean;
    'isOptInOutUnderstood': boolean;
    'isShortCodeUnderstood': boolean;
    'isStandardsUnderstood': boolean;
    'isCtiaShortCodeRead': boolean;
    'isManualRead': boolean;
    'isStopUnderstood': boolean;
    'isTosOnWebsite': boolean;
    'isPrivacyPolicyOnWebsite': boolean;
    'isTwilioTermsRead': boolean;

    static discriminator: string | undefined = undefined;

    static attributeTypeMap: Array<{name: string, baseName: string, type: string}> = [
        {
            "name": "isShortCodeTimelineUnderstood",
            "baseName": "isShortCodeTimelineUnderstood",
            "type": "boolean"
        },
        {
            "name": "isPricingUnderstood",
            "baseName": "isPricingUnderstood",
            "type": "boolean"
        },
        {
            "name": "isShortCodeTransferUnderstood",
            "baseName": "isShortCodeTransferUnderstood",
            "type": "boolean"
        },
        {
            "name": "isOptInOutUnderstood",
            "baseName": "isOptInOutUnderstood",
            "type": "boolean"
        },
        {
            "name": "isShortCodeUnderstood",
            "baseName": "isShortCodeUnderstood",
            "type": "boolean"
        },
        {
            "name": "isStandardsUnderstood",
            "baseName": "isStandardsUnderstood",
            "type": "boolean"
        },
        {
            "name": "isCtiaShortCodeRead",
            "baseName": "isCtiaShortCodeRead",
            "type": "boolean"
        },
        {
            "name": "isManualRead",
            "baseName": "isManualRead",
            "type": "boolean"
        },
        {
            "name": "isStopUnderstood",
            "baseName": "isStopUnderstood",
            "type": "boolean"
        },
        {
            "name": "isTosOnWebsite",
            "baseName": "isTosOnWebsite",
            "type": "boolean"
        },
        {
            "name": "isPrivacyPolicyOnWebsite",
            "baseName": "isPrivacyPolicyOnWebsite",
            "type": "boolean"
        },
        {
            "name": "isTwilioTermsRead",
            "baseName": "isTwilioTermsRead",
            "type": "boolean"
        }    ];

    static getAttributeTypeMap() {
        return InlineObject51.attributeTypeMap;
    }
}
