// wallet/model/inlineObject65.ts
/**
 * wallet-api
 * API
 *
 * The version of the OpenAPI document: 1.1.6
 * Contact: development@wallet.inc
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { RequestFile } from './models';

export class InlineObject65 {
    'googleAnalyticsID'?: string;
    'isChatRoom': boolean;
    'isSettings': boolean;
    'isProfile': boolean;
    'isTransactionHistory': boolean;
    'isLinkBook': boolean;
    'isCall': boolean;
    'isMessages': boolean;
    'isPerformances': boolean;
    'isNewsArticles': boolean;
    'isTickets'?: boolean;
    'isMerchantCredit': boolean;
    'isPromotions': boolean;
    'isGiftCertificates': boolean;
    'isGiftCards': boolean;
    'isMembershipLevel': boolean;
    'isMembershipPoints': boolean;
    'isMembershipTier': boolean;
    'isDynamicVouchers': boolean;
    'isStaticVouchers': boolean;
    'isAdCredits': boolean;
    'isSamsungEnabled': boolean;
    'isGoogleEnabled': boolean;
    'isAppleEnabled': boolean;
    'welcomeMessage': string;
    'companyLogoURL': string;
    'leftMenuArrowColor': string;
    'leftMenuSectionColor': string;
    'headerButtonColor': string;
    'headerBackgroundColor': string;

    static discriminator: string | undefined = undefined;

    static attributeTypeMap: Array<{name: string, baseName: string, type: string}> = [
        {
            "name": "googleAnalyticsID",
            "baseName": "googleAnalyticsID",
            "type": "string"
        },
        {
            "name": "isChatRoom",
            "baseName": "isChatRoom",
            "type": "boolean"
        },
        {
            "name": "isSettings",
            "baseName": "isSettings",
            "type": "boolean"
        },
        {
            "name": "isProfile",
            "baseName": "isProfile",
            "type": "boolean"
        },
        {
            "name": "isTransactionHistory",
            "baseName": "isTransactionHistory",
            "type": "boolean"
        },
        {
            "name": "isLinkBook",
            "baseName": "isLinkBook",
            "type": "boolean"
        },
        {
            "name": "isCall",
            "baseName": "isCall",
            "type": "boolean"
        },
        {
            "name": "isMessages",
            "baseName": "isMessages",
            "type": "boolean"
        },
        {
            "name": "isPerformances",
            "baseName": "isPerformances",
            "type": "boolean"
        },
        {
            "name": "isNewsArticles",
            "baseName": "isNewsArticles",
            "type": "boolean"
        },
        {
            "name": "isTickets",
            "baseName": "isTickets",
            "type": "boolean"
        },
        {
            "name": "isMerchantCredit",
            "baseName": "isMerchantCredit",
            "type": "boolean"
        },
        {
            "name": "isPromotions",
            "baseName": "isPromotions",
            "type": "boolean"
        },
        {
            "name": "isGiftCertificates",
            "baseName": "isGiftCertificates",
            "type": "boolean"
        },
        {
            "name": "isGiftCards",
            "baseName": "isGiftCards",
            "type": "boolean"
        },
        {
            "name": "isMembershipLevel",
            "baseName": "isMembershipLevel",
            "type": "boolean"
        },
        {
            "name": "isMembershipPoints",
            "baseName": "isMembershipPoints",
            "type": "boolean"
        },
        {
            "name": "isMembershipTier",
            "baseName": "isMembershipTier",
            "type": "boolean"
        },
        {
            "name": "isDynamicVouchers",
            "baseName": "isDynamicVouchers",
            "type": "boolean"
        },
        {
            "name": "isStaticVouchers",
            "baseName": "isStaticVouchers",
            "type": "boolean"
        },
        {
            "name": "isAdCredits",
            "baseName": "isAdCredits",
            "type": "boolean"
        },
        {
            "name": "isSamsungEnabled",
            "baseName": "isSamsungEnabled",
            "type": "boolean"
        },
        {
            "name": "isGoogleEnabled",
            "baseName": "isGoogleEnabled",
            "type": "boolean"
        },
        {
            "name": "isAppleEnabled",
            "baseName": "isAppleEnabled",
            "type": "boolean"
        },
        {
            "name": "welcomeMessage",
            "baseName": "welcomeMessage",
            "type": "string"
        },
        {
            "name": "companyLogoURL",
            "baseName": "companyLogoURL",
            "type": "string"
        },
        {
            "name": "leftMenuArrowColor",
            "baseName": "leftMenuArrowColor",
            "type": "string"
        },
        {
            "name": "leftMenuSectionColor",
            "baseName": "leftMenuSectionColor",
            "type": "string"
        },
        {
            "name": "headerButtonColor",
            "baseName": "headerButtonColor",
            "type": "string"
        },
        {
            "name": "headerBackgroundColor",
            "baseName": "headerBackgroundColor",
            "type": "string"
        }    ];

    static getAttributeTypeMap() {
        return InlineObject65.attributeTypeMap;
    }
}
