// wallet/model/merchant.ts
/**
 * wallet-api
 * API
 *
 * The version of the OpenAPI document: 1.1.6
 * Contact: development@wallet.inc
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { RequestFile } from './models';

export class Merchant {
    'id': string;
    'createdAt': Date;
    'updatedAt': Date;
    'companyName': string;
    'address1': string;
    'address2': string;
    'city': string;
    'state': string;
    'country': string;
    'phoneNumber': string;
    'zip': string;
    'industry': string;
    'infoGenesisPropertyID': string;
    'isFrozen': boolean;
    'billingContactEmployeeID': string;
    'stripeCustomerID': string;
    'isPaymentMethodProvided': boolean;
    'planNickname': string;

    static discriminator: string | undefined = undefined;

    static attributeTypeMap: Array<{name: string, baseName: string, type: string}> = [
        {
            "name": "id",
            "baseName": "id",
            "type": "string"
        },
        {
            "name": "createdAt",
            "baseName": "createdAt",
            "type": "Date"
        },
        {
            "name": "updatedAt",
            "baseName": "updatedAt",
            "type": "Date"
        },
        {
            "name": "companyName",
            "baseName": "companyName",
            "type": "string"
        },
        {
            "name": "address1",
            "baseName": "address1",
            "type": "string"
        },
        {
            "name": "address2",
            "baseName": "address2",
            "type": "string"
        },
        {
            "name": "city",
            "baseName": "city",
            "type": "string"
        },
        {
            "name": "state",
            "baseName": "state",
            "type": "string"
        },
        {
            "name": "country",
            "baseName": "country",
            "type": "string"
        },
        {
            "name": "phoneNumber",
            "baseName": "phoneNumber",
            "type": "string"
        },
        {
            "name": "zip",
            "baseName": "zip",
            "type": "string"
        },
        {
            "name": "industry",
            "baseName": "industry",
            "type": "string"
        },
        {
            "name": "infoGenesisPropertyID",
            "baseName": "infoGenesisPropertyID",
            "type": "string"
        },
        {
            "name": "isFrozen",
            "baseName": "isFrozen",
            "type": "boolean"
        },
        {
            "name": "billingContactEmployeeID",
            "baseName": "billingContactEmployeeID",
            "type": "string"
        },
        {
            "name": "stripeCustomerID",
            "baseName": "stripeCustomerID",
            "type": "string"
        },
        {
            "name": "isPaymentMethodProvided",
            "baseName": "isPaymentMethodProvided",
            "type": "boolean"
        },
        {
            "name": "planNickname",
            "baseName": "planNickname",
            "type": "string"
        }    ];

    static getAttributeTypeMap() {
        return Merchant.attributeTypeMap;
    }
}
