// wallet/model/models.ts
import localVarRequest from 'request';

export * from './advertisementCredit';
export * from './advertisementCreditBroadcast';
export * from './advertisementCreditScan';
export * from './agreement';
export * from './alert';
export * from './announcement';
export * from './applicableTerminals';
export * from './applicableTerminalsAnyOf';
export * from './applicableTerminalsAnyOf1';
export * from './applicableTerminalsAnyOf2';
export * from './applicableTerminalsAnyOf3';
export * from './authError';
export * from './availablePhoneNumbersRequest';
export * from './clickFunnelAmount';
export * from './clickFunnelContact';
export * from './clickFunnelContactProfile';
export * from './clickFunnelEvent';
export * from './clickFunnelOriginalAmount';
export * from './clickFunnelProduct';
export * from './clickFunnelPurchase';
export * from './createStaticVoucherCampaign';
export * from './createStaticVoucherCampaignWithVoucher';
export * from './createStaticVoucherCampaignWithVoucherWithCSV';
export * from './dashboardWidget';
export * from './duplicateRowFound';
export * from './dynamicVoucher';
export * from './dynamicVoucherBroadcast';
export * from './employee';
export * from './employeeActivityLog';
export * from './employeeAlert';
export * from './exportDataFile';
export * from './falsumError';
export * from './falsumField';
export * from './feature';
export * from './forbiddenRequest';
export * from './foreignKeyDoesNotExist';
export * from './helpDeskRequest';
export * from './importedList';
export * from './importedListRecipient';
export * from './inlineObject';
export * from './inlineObject1';
export * from './inlineObject10';
export * from './inlineObject11';
export * from './inlineObject12';
export * from './inlineObject13';
export * from './inlineObject14';
export * from './inlineObject15';
export * from './inlineObject16';
export * from './inlineObject17';
export * from './inlineObject18';
export * from './inlineObject19';
export * from './inlineObject2';
export * from './inlineObject20';
export * from './inlineObject21';
export * from './inlineObject22';
export * from './inlineObject23';
export * from './inlineObject24';
export * from './inlineObject25';
export * from './inlineObject26';
export * from './inlineObject27';
export * from './inlineObject28';
export * from './inlineObject29';
export * from './inlineObject3';
export * from './inlineObject30';
export * from './inlineObject31';
export * from './inlineObject32';
export * from './inlineObject33';
export * from './inlineObject34';
export * from './inlineObject35';
export * from './inlineObject36';
export * from './inlineObject37';
export * from './inlineObject38';
export * from './inlineObject39';
export * from './inlineObject4';
export * from './inlineObject40';
export * from './inlineObject41';
export * from './inlineObject42';
export * from './inlineObject43';
export * from './inlineObject44';
export * from './inlineObject45';
export * from './inlineObject46';
export * from './inlineObject47';
export * from './inlineObject48';
export * from './inlineObject49';
export * from './inlineObject5';
export * from './inlineObject50';
export * from './inlineObject51';
export * from './inlineObject52';
export * from './inlineObject53';
export * from './inlineObject54';
export * from './inlineObject55';
export * from './inlineObject56';
export * from './inlineObject57';
export * from './inlineObject58';
export * from './inlineObject59';
export * from './inlineObject6';
export * from './inlineObject60';
export * from './inlineObject61';
export * from './inlineObject62';
export * from './inlineObject63';
export * from './inlineObject64';
export * from './inlineObject65';
export * from './inlineObject66';
export * from './inlineObject67';
export * from './inlineObject68';
export * from './inlineObject69';
export * from './inlineObject7';
export * from './inlineObject8';
export * from './inlineObject9';
export * from './inlineResponse200';
export * from './inlineResponse2001';
export * from './inlineResponse20010';
export * from './inlineResponse20011';
export * from './inlineResponse2002';
export * from './inlineResponse2003';
export * from './inlineResponse2004';
export * from './inlineResponse2005';
export * from './inlineResponse2006';
export * from './inlineResponse2007';
export * from './inlineResponse2008';
export * from './inlineResponse2009';
export * from './internalServerError';
export * from './ledgerEntry';
export * from './linkBook';
export * from './linkBookSection';
export * from './mSMemberHistory';
export * from './mSMemberHistoryPagination';
export * from './mSMemberRedemption';
export * from './mSMemberRedemptionPagination';
export * from './mSMembershipTierHistory';
export * from './mSMembershipTierHistoryPagination';
export * from './mSMembershipTierRedemption';
export * from './mSMembershipTierRedemptionPagination';
export * from './mSMerchantCreditHistory';
export * from './mSMerchantCreditHistoryPagination';
export * from './mSMerchantCreditRedemption';
export * from './mSMerchantCreditRedemptionPagination';
export * from './mediaFile';
export * from './member';
export * from './memberSearch';
export * from './merchant';
export * from './merchantCreditSearch';
export * from './merchantNotInitialized';
export * from './merchantURL';
export * from './message';
export * from './moduleError';
export * from './newsArticle';
export * from './optInList';
export * from './optInListSource';
export * from './optInListSubscriber';
export * from './outboundSMS';
export * from './paginatedWTMembers';
export * from './paginatedWTMerchantCredits';
export * from './paginationRequestWithSortOptions';
export * from './paymentDesign';
export * from './performance';
export * from './phoneNumber';
export * from './pickCreateStaticVoucherCampaignWithVoucherExcludeKeyofcreateStaticVoucherCampaignWithVoucherSourceID';
export * from './pickMSMemberCreationParamsExcludeKeyofMSMemberCreationParamsMemberIdentifier';
export * from './pickMSMemberExcludeKeyofMSMemberMemberIdentifier';
export * from './pickMSMerchantCreditCreationParamsExcludeKeyofMSMerchantCreditCreationParamsMemberIdentifier';
export * from './pickMSMerchantCreditExcludeKeyofMSMerchantCreditMemberIdentifier';
export * from './pickPaginationRequestWithoutSortOptionsExcludeKeyofPaginationRequestWithoutSortOptionsIsArchiveIncluded';
export * from './pickSSImportedListUpdateParamsExcludeKeyofSSImportedListUpdateParamsId';
export * from './pickSSMobileNumberUpdateParamsExcludeKeyofSSMobileNumberUpdateParamsId';
export * from './pickSSOptInListCreateParamsExcludeKeyofSSOptInListCreateParamsEmployeeID';
export * from './pickSSOptInListUpdateParamsExcludeKeyofSSOptInListUpdateParamsId';
export * from './pickSSOptInSourceUpdateParamsExcludeKeyofSSOptInSourceUpdateParamsId';
export * from './pickSSOutboundMessageLogExcludeKeyofSSOutboundMessageLogToCellPhone';
export * from './pickVSAdvertisementCreditScanExcludeKeyofVSAdvertisementCreditScanRedeemedAtOrRefundedAt';
export * from './pickVSAdvertisementCreditUpdateParamsExcludeKeyofVSAdvertisementCreditUpdateParamsId';
export * from './pickVSCampaignUpdateParamsExcludeKeyofVSCampaignUpdateParamsStartDateOrExpirationDate';
export * from './pickVSDynamicVoucherExcludeKeyofVSDynamicVoucherTemporalDecreaseFrequencyType';
export * from './pickVSPaymentDesignCreateParamsExcludeKeyofVSPaymentDesignCreateParamsBackgroundImageURLOrCompanyLogoURLOrEmployeeID';
export * from './pickVSPaymentDesignUpdateParamsExcludeKeyofVSPaymentDesignUpdateParamsId';
export * from './pickVSStaticVoucherExcludeKeyofVSStaticVoucherRedeemedAtOrRefundedAtOrLastViewedAt';
export * from './pickWTAdvertisementCreditCreateParamsExcludeKeyofWTAdvertisementCreditCreateParamsEmployeeID';
export * from './pickWTEmployeeCreateExcludeKeyofWTEmployeeCreateEmail';
export * from './pickWTMemberMemberIDOrFirstNameOrLastNameOrMembershipTierIDOrPointsAccruedOrMobileNumber';
export * from './pickWTMerchantCreditMemberIDOrCreditAmountOrMobileNumber';
export * from './pickWTMerchantURLUpdateExcludeKeyofWTMerchantURLUpdateUrlID';
export * from './pickWTNewsArticleUpdateParamsExcludeKeyofWTNewsArticleUpdateParamsId';
export * from './pickWTPaymentObjectBroadcastExcludeKeyofWTPaymentObjectBroadcastListTypeOrListID';
export * from './pickWTStaticVoucherCampaignCreateParamsExcludeKeyofWTStaticVoucherCampaignCreateParamsStartDateTimeOrExpirationDateTime';
export * from './pickWTStaticVoucherCreateParamsExcludeKeyofWTStaticVoucherCreateParamsCampaignID';
export * from './plan';
export * from './portalPage';
export * from './presignedPost';
export * from './presignedPostFields';
export * from './product';
export * from './promoCode';
export * from './reachPerformanceStats';
export * from './request';
export * from './response';
export * from './role';
export * from './roleAuditLog';
export * from './sSOptInSource';
export * from './simpleSMSBroadcast';
export * from './staticVoucher';
export * from './staticVoucherCampaign';
export * from './staticVoucherCampaignBroadcast';
export * from './staticVoucherCampaignGroup';
export * from './staticVoucherCampaignUpdate';
export * from './status';
export * from './statusAnyOf';
export * from './statusAnyOf1';
export * from './statusAnyOf2';
export * from './statusAnyOf3';
export * from './subscriptionFeature';
export * from './subscriptionPlan';
export * from './subscriptionProduct';
export * from './tcpa';
export * from './updateStaticVoucherCampaignWithVoucher';
export * from './vSCampaignGeneratedMessage';
export * from './vSCampaignGeneratedMessagePagination';
export * from './vSDynamicVoucherStatus';
export * from './wTAdvertisementCredit';
export * from './wTAdvertisementCreditScan';
export * from './wTAdvertisementCreditUpdateParams';
export * from './wTDynamicVoucher';
export * from './wTDynamicVoucherCreateParams';
export * from './wTDynamicVoucherRedemption';
export * from './wTDynamicVoucherSummary';
export * from './wTDynamicVoucherUpdateParams';
export * from './wTEmployee';
export * from './wTEmployeeCreate';
export * from './wTEmployeeUpdate';
export * from './wTImportedList';
export * from './wTLinkBookCreateParams';
export * from './wTLinkBookSectionCreateParams';
export * from './wTLinkBookSectionUpdateParams';
export * from './wTLinkBookUpdateParams';
export * from './wTMember';
export * from './wTMemberCreationParams';
export * from './wTMembershipTier';
export * from './wTMembershipTierCreationParams';
export * from './wTMembershipTierUpdateParams';
export * from './wTMembershipTierWithMemberCount';
export * from './wTMerchantCredit';
export * from './wTMerchantCreditCreationParams';
export * from './wTMerchantURLCreate';
export * from './wTMerchantURLUpdate';
export * from './wTNewsArticleCreateParams';
export * from './wTNewsArticleUpdateParams';
export * from './wTOptInList';
export * from './wTOptInListCreationParams';
export * from './wTPaymentDesign';
export * from './wTPaymentDesignCreateParams';
export * from './wTPaymentDesignUpdateParams';
export * from './wTPerformanceCreateParams';
export * from './wTPerformanceUpdateParams';
export * from './wTPosMachine';
export * from './wTPosMachineCreateParams';
export * from './wTPosMachineUpdateParams';
export * from './wTPromoCodeCreateParams';
export * from './wTPromoCodeUpdateParams';
export * from './wTRole';
export * from './wTStaticVoucher';
export * from './wTStaticVoucherCampaign';
export * from './wTStaticVoucherCreateParams';
export * from './wTStaticVoucherUpdateParams';
export * from './wTWalletPageView';
export * from './wTWalletPageViewGeoPoint';
export * from './walletPageView';
export * from './webpage';

import * as fs from 'fs';

export interface RequestDetailedFile {
    value: Buffer;
    options?: {
        filename?: string;
        contentType?: string;
    }
}

export type RequestFile = string | Buffer | fs.ReadStream | RequestDetailedFile;


import { AdvertisementCredit } from './advertisementCredit';
import { AdvertisementCreditBroadcast } from './advertisementCreditBroadcast';
import { AdvertisementCreditScan } from './advertisementCreditScan';
import { Agreement } from './agreement';
import { Alert } from './alert';
import { Announcement } from './announcement';
import { ApplicableTerminals } from './applicableTerminals';
import { ApplicableTerminalsAnyOf } from './applicableTerminalsAnyOf';
import { ApplicableTerminalsAnyOf1 } from './applicableTerminalsAnyOf1';
import { ApplicableTerminalsAnyOf2 } from './applicableTerminalsAnyOf2';
import { ApplicableTerminalsAnyOf3 } from './applicableTerminalsAnyOf3';
import { AuthError } from './authError';
import { AvailablePhoneNumbersRequest } from './availablePhoneNumbersRequest';
import { ClickFunnelAmount } from './clickFunnelAmount';
import { ClickFunnelContact } from './clickFunnelContact';
import { ClickFunnelContactProfile } from './clickFunnelContactProfile';
import { ClickFunnelEvent } from './clickFunnelEvent';
import { ClickFunnelOriginalAmount } from './clickFunnelOriginalAmount';
import { ClickFunnelProduct } from './clickFunnelProduct';
import { ClickFunnelPurchase } from './clickFunnelPurchase';
import { CreateStaticVoucherCampaign } from './createStaticVoucherCampaign';
import { CreateStaticVoucherCampaignWithVoucher } from './createStaticVoucherCampaignWithVoucher';
import { CreateStaticVoucherCampaignWithVoucherWithCSV } from './createStaticVoucherCampaignWithVoucherWithCSV';
import { DashboardWidget } from './dashboardWidget';
import { DuplicateRowFound } from './duplicateRowFound';
import { DynamicVoucher } from './dynamicVoucher';
import { DynamicVoucherBroadcast } from './dynamicVoucherBroadcast';
import { Employee } from './employee';
import { EmployeeActivityLog } from './employeeActivityLog';
import { EmployeeAlert } from './employeeAlert';
import { ExportDataFile } from './exportDataFile';
import { FalsumError } from './falsumError';
import { FalsumField } from './falsumField';
import { Feature } from './feature';
import { ForbiddenRequest } from './forbiddenRequest';
import { ForeignKeyDoesNotExist } from './foreignKeyDoesNotExist';
import { HelpDeskRequest } from './helpDeskRequest';
import { ImportedList } from './importedList';
import { ImportedListRecipient } from './importedListRecipient';
import { InlineObject } from './inlineObject';
import { InlineObject1 } from './inlineObject1';
import { InlineObject10 } from './inlineObject10';
import { InlineObject11 } from './inlineObject11';
import { InlineObject12 } from './inlineObject12';
import { InlineObject13 } from './inlineObject13';
import { InlineObject14 } from './inlineObject14';
import { InlineObject15 } from './inlineObject15';
import { InlineObject16 } from './inlineObject16';
import { InlineObject17 } from './inlineObject17';
import { InlineObject18 } from './inlineObject18';
import { InlineObject19 } from './inlineObject19';
import { InlineObject2 } from './inlineObject2';
import { InlineObject20 } from './inlineObject20';
import { InlineObject21 } from './inlineObject21';
import { InlineObject22 } from './inlineObject22';
import { InlineObject23 } from './inlineObject23';
import { InlineObject24 } from './inlineObject24';
import { InlineObject25 } from './inlineObject25';
import { InlineObject26 } from './inlineObject26';
import { InlineObject27 } from './inlineObject27';
import { InlineObject28 } from './inlineObject28';
import { InlineObject29 } from './inlineObject29';
import { InlineObject3 } from './inlineObject3';
import { InlineObject30 } from './inlineObject30';
import { InlineObject31 } from './inlineObject31';
import { InlineObject32 } from './inlineObject32';
import { InlineObject33 } from './inlineObject33';
import { InlineObject34 } from './inlineObject34';
import { InlineObject35 } from './inlineObject35';
import { InlineObject36 } from './inlineObject36';
import { InlineObject37 } from './inlineObject37';
import { InlineObject38 } from './inlineObject38';
import { InlineObject39 } from './inlineObject39';
import { InlineObject4 } from './inlineObject4';
import { InlineObject40 } from './inlineObject40';
import { InlineObject41 } from './inlineObject41';
import { InlineObject42 } from './inlineObject42';
import { InlineObject43 } from './inlineObject43';
import { InlineObject44 } from './inlineObject44';
import { InlineObject45 } from './inlineObject45';
import { InlineObject46 } from './inlineObject46';
import { InlineObject47 } from './inlineObject47';
import { InlineObject48 } from './inlineObject48';
import { InlineObject49 } from './inlineObject49';
import { InlineObject5 } from './inlineObject5';
import { InlineObject50 } from './inlineObject50';
import { InlineObject51 } from './inlineObject51';
import { InlineObject52 } from './inlineObject52';
import { InlineObject53 } from './inlineObject53';
import { InlineObject54 } from './inlineObject54';
import { InlineObject55 } from './inlineObject55';
import { InlineObject56 } from './inlineObject56';
import { InlineObject57 } from './inlineObject57';
import { InlineObject58 } from './inlineObject58';
import { InlineObject59 } from './inlineObject59';
import { InlineObject6 } from './inlineObject6';
import { InlineObject60 } from './inlineObject60';
import { InlineObject61 } from './inlineObject61';
import { InlineObject62 } from './inlineObject62';
import { InlineObject63 } from './inlineObject63';
import { InlineObject64 } from './inlineObject64';
import { InlineObject65 } from './inlineObject65';
import { InlineObject66 } from './inlineObject66';
import { InlineObject67 } from './inlineObject67';
import { InlineObject68 } from './inlineObject68';
import { InlineObject69 } from './inlineObject69';
import { InlineObject7 } from './inlineObject7';
import { InlineObject8 } from './inlineObject8';
import { InlineObject9 } from './inlineObject9';
import { InlineResponse200 } from './inlineResponse200';
import { InlineResponse2001 } from './inlineResponse2001';
import { InlineResponse20010 } from './inlineResponse20010';
import { InlineResponse20011 } from './inlineResponse20011';
import { InlineResponse2002 } from './inlineResponse2002';
import { InlineResponse2003 } from './inlineResponse2003';
import { InlineResponse2004 } from './inlineResponse2004';
import { InlineResponse2005 } from './inlineResponse2005';
import { InlineResponse2006 } from './inlineResponse2006';
import { InlineResponse2007 } from './inlineResponse2007';
import { InlineResponse2008 } from './inlineResponse2008';
import { InlineResponse2009 } from './inlineResponse2009';
import { InternalServerError } from './internalServerError';
import { LedgerEntry } from './ledgerEntry';
import { LinkBook } from './linkBook';
import { LinkBookSection } from './linkBookSection';
import { MSMemberHistory } from './mSMemberHistory';
import { MSMemberHistoryPagination } from './mSMemberHistoryPagination';
import { MSMemberRedemption } from './mSMemberRedemption';
import { MSMemberRedemptionPagination } from './mSMemberRedemptionPagination';
import { MSMembershipTierHistory } from './mSMembershipTierHistory';
import { MSMembershipTierHistoryPagination } from './mSMembershipTierHistoryPagination';
import { MSMembershipTierRedemption } from './mSMembershipTierRedemption';
import { MSMembershipTierRedemptionPagination } from './mSMembershipTierRedemptionPagination';
import { MSMerchantCreditHistory } from './mSMerchantCreditHistory';
import { MSMerchantCreditHistoryPagination } from './mSMerchantCreditHistoryPagination';
import { MSMerchantCreditRedemption } from './mSMerchantCreditRedemption';
import { MSMerchantCreditRedemptionPagination } from './mSMerchantCreditRedemptionPagination';
import { MediaFile } from './mediaFile';
import { Member } from './member';
import { MemberSearch } from './memberSearch';
import { Merchant } from './merchant';
import { MerchantCreditSearch } from './merchantCreditSearch';
import { MerchantNotInitialized } from './merchantNotInitialized';
import { MerchantURL } from './merchantURL';
import { Message } from './message';
import { ModuleError } from './moduleError';
import { NewsArticle } from './newsArticle';
import { OptInList } from './optInList';
import { OptInListSource } from './optInListSource';
import { OptInListSubscriber } from './optInListSubscriber';
import { OutboundSMS } from './outboundSMS';
import { PaginatedWTMembers } from './paginatedWTMembers';
import { PaginatedWTMerchantCredits } from './paginatedWTMerchantCredits';
import { PaginationRequestWithSortOptions } from './paginationRequestWithSortOptions';
import { PaymentDesign } from './paymentDesign';
import { Performance } from './performance';
import { PhoneNumber } from './phoneNumber';
import { PickCreateStaticVoucherCampaignWithVoucherExcludeKeyofcreateStaticVoucherCampaignWithVoucherSourceID } from './pickCreateStaticVoucherCampaignWithVoucherExcludeKeyofcreateStaticVoucherCampaignWithVoucherSourceID';
import { PickMSMemberCreationParamsExcludeKeyofMSMemberCreationParamsMemberIdentifier } from './pickMSMemberCreationParamsExcludeKeyofMSMemberCreationParamsMemberIdentifier';
import { PickMSMemberExcludeKeyofMSMemberMemberIdentifier } from './pickMSMemberExcludeKeyofMSMemberMemberIdentifier';
import { PickMSMerchantCreditCreationParamsExcludeKeyofMSMerchantCreditCreationParamsMemberIdentifier } from './pickMSMerchantCreditCreationParamsExcludeKeyofMSMerchantCreditCreationParamsMemberIdentifier';
import { PickMSMerchantCreditExcludeKeyofMSMerchantCreditMemberIdentifier } from './pickMSMerchantCreditExcludeKeyofMSMerchantCreditMemberIdentifier';
import { PickPaginationRequestWithoutSortOptionsExcludeKeyofPaginationRequestWithoutSortOptionsIsArchiveIncluded } from './pickPaginationRequestWithoutSortOptionsExcludeKeyofPaginationRequestWithoutSortOptionsIsArchiveIncluded';
import { PickSSImportedListUpdateParamsExcludeKeyofSSImportedListUpdateParamsId } from './pickSSImportedListUpdateParamsExcludeKeyofSSImportedListUpdateParamsId';
import { PickSSMobileNumberUpdateParamsExcludeKeyofSSMobileNumberUpdateParamsId } from './pickSSMobileNumberUpdateParamsExcludeKeyofSSMobileNumberUpdateParamsId';
import { PickSSOptInListCreateParamsExcludeKeyofSSOptInListCreateParamsEmployeeID } from './pickSSOptInListCreateParamsExcludeKeyofSSOptInListCreateParamsEmployeeID';
import { PickSSOptInListUpdateParamsExcludeKeyofSSOptInListUpdateParamsId } from './pickSSOptInListUpdateParamsExcludeKeyofSSOptInListUpdateParamsId';
import { PickSSOptInSourceUpdateParamsExcludeKeyofSSOptInSourceUpdateParamsId } from './pickSSOptInSourceUpdateParamsExcludeKeyofSSOptInSourceUpdateParamsId';
import { PickSSOutboundMessageLogExcludeKeyofSSOutboundMessageLogToCellPhone } from './pickSSOutboundMessageLogExcludeKeyofSSOutboundMessageLogToCellPhone';
import { PickVSAdvertisementCreditScanExcludeKeyofVSAdvertisementCreditScanRedeemedAtOrRefundedAt } from './pickVSAdvertisementCreditScanExcludeKeyofVSAdvertisementCreditScanRedeemedAtOrRefundedAt';
import { PickVSAdvertisementCreditUpdateParamsExcludeKeyofVSAdvertisementCreditUpdateParamsId } from './pickVSAdvertisementCreditUpdateParamsExcludeKeyofVSAdvertisementCreditUpdateParamsId';
import { PickVSCampaignUpdateParamsExcludeKeyofVSCampaignUpdateParamsStartDateOrExpirationDate } from './pickVSCampaignUpdateParamsExcludeKeyofVSCampaignUpdateParamsStartDateOrExpirationDate';
import { PickVSDynamicVoucherExcludeKeyofVSDynamicVoucherTemporalDecreaseFrequencyType } from './pickVSDynamicVoucherExcludeKeyofVSDynamicVoucherTemporalDecreaseFrequencyType';
import { PickVSPaymentDesignCreateParamsExcludeKeyofVSPaymentDesignCreateParamsBackgroundImageURLOrCompanyLogoURLOrEmployeeID } from './pickVSPaymentDesignCreateParamsExcludeKeyofVSPaymentDesignCreateParamsBackgroundImageURLOrCompanyLogoURLOrEmployeeID';
import { PickVSPaymentDesignUpdateParamsExcludeKeyofVSPaymentDesignUpdateParamsId } from './pickVSPaymentDesignUpdateParamsExcludeKeyofVSPaymentDesignUpdateParamsId';
import { PickVSStaticVoucherExcludeKeyofVSStaticVoucherRedeemedAtOrRefundedAtOrLastViewedAt } from './pickVSStaticVoucherExcludeKeyofVSStaticVoucherRedeemedAtOrRefundedAtOrLastViewedAt';
import { PickWTAdvertisementCreditCreateParamsExcludeKeyofWTAdvertisementCreditCreateParamsEmployeeID } from './pickWTAdvertisementCreditCreateParamsExcludeKeyofWTAdvertisementCreditCreateParamsEmployeeID';
import { PickWTEmployeeCreateExcludeKeyofWTEmployeeCreateEmail } from './pickWTEmployeeCreateExcludeKeyofWTEmployeeCreateEmail';
import { PickWTMemberMemberIDOrFirstNameOrLastNameOrMembershipTierIDOrPointsAccruedOrMobileNumber } from './pickWTMemberMemberIDOrFirstNameOrLastNameOrMembershipTierIDOrPointsAccruedOrMobileNumber';
import { PickWTMerchantCreditMemberIDOrCreditAmountOrMobileNumber } from './pickWTMerchantCreditMemberIDOrCreditAmountOrMobileNumber';
import { PickWTMerchantURLUpdateExcludeKeyofWTMerchantURLUpdateUrlID } from './pickWTMerchantURLUpdateExcludeKeyofWTMerchantURLUpdateUrlID';
import { PickWTNewsArticleUpdateParamsExcludeKeyofWTNewsArticleUpdateParamsId } from './pickWTNewsArticleUpdateParamsExcludeKeyofWTNewsArticleUpdateParamsId';
import { PickWTPaymentObjectBroadcastExcludeKeyofWTPaymentObjectBroadcastListTypeOrListID } from './pickWTPaymentObjectBroadcastExcludeKeyofWTPaymentObjectBroadcastListTypeOrListID';
import { PickWTStaticVoucherCampaignCreateParamsExcludeKeyofWTStaticVoucherCampaignCreateParamsStartDateTimeOrExpirationDateTime } from './pickWTStaticVoucherCampaignCreateParamsExcludeKeyofWTStaticVoucherCampaignCreateParamsStartDateTimeOrExpirationDateTime';
import { PickWTStaticVoucherCreateParamsExcludeKeyofWTStaticVoucherCreateParamsCampaignID } from './pickWTStaticVoucherCreateParamsExcludeKeyofWTStaticVoucherCreateParamsCampaignID';
import { Plan } from './plan';
import { PortalPage } from './portalPage';
import { PresignedPost } from './presignedPost';
import { PresignedPostFields } from './presignedPostFields';
import { Product } from './product';
import { PromoCode } from './promoCode';
import { ReachPerformanceStats } from './reachPerformanceStats';
import { Request } from './request';
import { Response } from './response';
import { Role } from './role';
import { RoleAuditLog } from './roleAuditLog';
import { SSOptInSource } from './sSOptInSource';
import { SimpleSMSBroadcast } from './simpleSMSBroadcast';
import { StaticVoucher } from './staticVoucher';
import { StaticVoucherCampaign } from './staticVoucherCampaign';
import { StaticVoucherCampaignBroadcast } from './staticVoucherCampaignBroadcast';
import { StaticVoucherCampaignGroup } from './staticVoucherCampaignGroup';
import { StaticVoucherCampaignUpdate } from './staticVoucherCampaignUpdate';
import { Status } from './status';
import { StatusAnyOf } from './statusAnyOf';
import { StatusAnyOf1 } from './statusAnyOf1';
import { StatusAnyOf2 } from './statusAnyOf2';
import { StatusAnyOf3 } from './statusAnyOf3';
import { SubscriptionFeature } from './subscriptionFeature';
import { SubscriptionPlan } from './subscriptionPlan';
import { SubscriptionProduct } from './subscriptionProduct';
import { Tcpa } from './tcpa';
import { UpdateStaticVoucherCampaignWithVoucher } from './updateStaticVoucherCampaignWithVoucher';
import { VSCampaignGeneratedMessage } from './vSCampaignGeneratedMessage';
import { VSCampaignGeneratedMessagePagination } from './vSCampaignGeneratedMessagePagination';
import { VSDynamicVoucherStatus } from './vSDynamicVoucherStatus';
import { WTAdvertisementCredit } from './wTAdvertisementCredit';
import { WTAdvertisementCreditScan } from './wTAdvertisementCreditScan';
import { WTAdvertisementCreditUpdateParams } from './wTAdvertisementCreditUpdateParams';
import { WTDynamicVoucher } from './wTDynamicVoucher';
import { WTDynamicVoucherCreateParams } from './wTDynamicVoucherCreateParams';
import { WTDynamicVoucherRedemption } from './wTDynamicVoucherRedemption';
import { WTDynamicVoucherSummary } from './wTDynamicVoucherSummary';
import { WTDynamicVoucherUpdateParams } from './wTDynamicVoucherUpdateParams';
import { WTEmployee } from './wTEmployee';
import { WTEmployeeCreate } from './wTEmployeeCreate';
import { WTEmployeeUpdate } from './wTEmployeeUpdate';
import { WTImportedList } from './wTImportedList';
import { WTLinkBookCreateParams } from './wTLinkBookCreateParams';
import { WTLinkBookSectionCreateParams } from './wTLinkBookSectionCreateParams';
import { WTLinkBookSectionUpdateParams } from './wTLinkBookSectionUpdateParams';
import { WTLinkBookUpdateParams } from './wTLinkBookUpdateParams';
import { WTMember } from './wTMember';
import { WTMemberCreationParams } from './wTMemberCreationParams';
import { WTMembershipTier } from './wTMembershipTier';
import { WTMembershipTierCreationParams } from './wTMembershipTierCreationParams';
import { WTMembershipTierUpdateParams } from './wTMembershipTierUpdateParams';
import { WTMembershipTierWithMemberCount } from './wTMembershipTierWithMemberCount';
import { WTMerchantCredit } from './wTMerchantCredit';
import { WTMerchantCreditCreationParams } from './wTMerchantCreditCreationParams';
import { WTMerchantURLCreate } from './wTMerchantURLCreate';
import { WTMerchantURLUpdate } from './wTMerchantURLUpdate';
import { WTNewsArticleCreateParams } from './wTNewsArticleCreateParams';
import { WTNewsArticleUpdateParams } from './wTNewsArticleUpdateParams';
import { WTOptInList } from './wTOptInList';
import { WTOptInListCreationParams } from './wTOptInListCreationParams';
import { WTPaymentDesign } from './wTPaymentDesign';
import { WTPaymentDesignCreateParams } from './wTPaymentDesignCreateParams';
import { WTPaymentDesignUpdateParams } from './wTPaymentDesignUpdateParams';
import { WTPerformanceCreateParams } from './wTPerformanceCreateParams';
import { WTPerformanceUpdateParams } from './wTPerformanceUpdateParams';
import { WTPosMachine } from './wTPosMachine';
import { WTPosMachineCreateParams } from './wTPosMachineCreateParams';
import { WTPosMachineUpdateParams } from './wTPosMachineUpdateParams';
import { WTPromoCodeCreateParams } from './wTPromoCodeCreateParams';
import { WTPromoCodeUpdateParams } from './wTPromoCodeUpdateParams';
import { WTRole } from './wTRole';
import { WTStaticVoucher } from './wTStaticVoucher';
import { WTStaticVoucherCampaign } from './wTStaticVoucherCampaign';
import { WTStaticVoucherCreateParams } from './wTStaticVoucherCreateParams';
import { WTStaticVoucherUpdateParams } from './wTStaticVoucherUpdateParams';
import { WTWalletPageView } from './wTWalletPageView';
import { WTWalletPageViewGeoPoint } from './wTWalletPageViewGeoPoint';
import { WalletPageView } from './walletPageView';
import { Webpage } from './webpage';

/* tslint:disable:no-unused-variable */
let primitives = [
                    "string",
                    "boolean",
                    "double",
                    "integer",
                    "long",
                    "float",
                    "number",
                    "any"
                 ];

let enumsMap: {[index: string]: any} = {
        "ApplicableTerminalsAnyOf": ApplicableTerminalsAnyOf,
        "ApplicableTerminalsAnyOf1": ApplicableTerminalsAnyOf1,
        "ApplicableTerminalsAnyOf2": ApplicableTerminalsAnyOf2,
        "ApplicableTerminalsAnyOf3": ApplicableTerminalsAnyOf3,
        "PaginationRequestWithSortOptions.SortKeyEnum": PaginationRequestWithSortOptions.SortKeyEnum,
        "PortalPage": PortalPage,
        "StatusAnyOf": StatusAnyOf,
        "StatusAnyOf1": StatusAnyOf1,
        "StatusAnyOf2": StatusAnyOf2,
        "StatusAnyOf3": StatusAnyOf3,
}

let typeMap: {[index: string]: any} = {
    "AdvertisementCredit": AdvertisementCredit,
    "AdvertisementCreditBroadcast": AdvertisementCreditBroadcast,
    "AdvertisementCreditScan": AdvertisementCreditScan,
    "Agreement": Agreement,
    "Alert": Alert,
    "Announcement": Announcement,
    "ApplicableTerminals": ApplicableTerminals,
    "AuthError": AuthError,
    "AvailablePhoneNumbersRequest": AvailablePhoneNumbersRequest,
    "ClickFunnelAmount": ClickFunnelAmount,
    "ClickFunnelContact": ClickFunnelContact,
    "ClickFunnelContactProfile": ClickFunnelContactProfile,
    "ClickFunnelEvent": ClickFunnelEvent,
    "ClickFunnelOriginalAmount": ClickFunnelOriginalAmount,
    "ClickFunnelProduct": ClickFunnelProduct,
    "ClickFunnelPurchase": ClickFunnelPurchase,
    "CreateStaticVoucherCampaign": CreateStaticVoucherCampaign,
    "CreateStaticVoucherCampaignWithVoucher": CreateStaticVoucherCampaignWithVoucher,
    "CreateStaticVoucherCampaignWithVoucherWithCSV": CreateStaticVoucherCampaignWithVoucherWithCSV,
    "DashboardWidget": DashboardWidget,
    "DuplicateRowFound": DuplicateRowFound,
    "DynamicVoucher": DynamicVoucher,
    "DynamicVoucherBroadcast": DynamicVoucherBroadcast,
    "Employee": Employee,
    "EmployeeActivityLog": EmployeeActivityLog,
    "EmployeeAlert": EmployeeAlert,
    "ExportDataFile": ExportDataFile,
    "FalsumError": FalsumError,
    "FalsumField": FalsumField,
    "Feature": Feature,
    "ForbiddenRequest": ForbiddenRequest,
    "ForeignKeyDoesNotExist": ForeignKeyDoesNotExist,
    "HelpDeskRequest": HelpDeskRequest,
    "ImportedList": ImportedList,
    "ImportedListRecipient": ImportedListRecipient,
    "InlineObject": InlineObject,
    "InlineObject1": InlineObject1,
    "InlineObject10": InlineObject10,
    "InlineObject11": InlineObject11,
    "InlineObject12": InlineObject12,
    "InlineObject13": InlineObject13,
    "InlineObject14": InlineObject14,
    "InlineObject15": InlineObject15,
    "InlineObject16": InlineObject16,
    "InlineObject17": InlineObject17,
    "InlineObject18": InlineObject18,
    "InlineObject19": InlineObject19,
    "InlineObject2": InlineObject2,
    "InlineObject20": InlineObject20,
    "InlineObject21": InlineObject21,
    "InlineObject22": InlineObject22,
    "InlineObject23": InlineObject23,
    "InlineObject24": InlineObject24,
    "InlineObject25": InlineObject25,
    "InlineObject26": InlineObject26,
    "InlineObject27": InlineObject27,
    "InlineObject28": InlineObject28,
    "InlineObject29": InlineObject29,
    "InlineObject3": InlineObject3,
    "InlineObject30": InlineObject30,
    "InlineObject31": InlineObject31,
    "InlineObject32": InlineObject32,
    "InlineObject33": InlineObject33,
    "InlineObject34": InlineObject34,
    "InlineObject35": InlineObject35,
    "InlineObject36": InlineObject36,
    "InlineObject37": InlineObject37,
    "InlineObject38": InlineObject38,
    "InlineObject39": InlineObject39,
    "InlineObject4": InlineObject4,
    "InlineObject40": InlineObject40,
    "InlineObject41": InlineObject41,
    "InlineObject42": InlineObject42,
    "InlineObject43": InlineObject43,
    "InlineObject44": InlineObject44,
    "InlineObject45": InlineObject45,
    "InlineObject46": InlineObject46,
    "InlineObject47": InlineObject47,
    "InlineObject48": InlineObject48,
    "InlineObject49": InlineObject49,
    "InlineObject5": InlineObject5,
    "InlineObject50": InlineObject50,
    "InlineObject51": InlineObject51,
    "InlineObject52": InlineObject52,
    "InlineObject53": InlineObject53,
    "InlineObject54": InlineObject54,
    "InlineObject55": InlineObject55,
    "InlineObject56": InlineObject56,
    "InlineObject57": InlineObject57,
    "InlineObject58": InlineObject58,
    "InlineObject59": InlineObject59,
    "InlineObject6": InlineObject6,
    "InlineObject60": InlineObject60,
    "InlineObject61": InlineObject61,
    "InlineObject62": InlineObject62,
    "InlineObject63": InlineObject63,
    "InlineObject64": InlineObject64,
    "InlineObject65": InlineObject65,
    "InlineObject66": InlineObject66,
    "InlineObject67": InlineObject67,
    "InlineObject68": InlineObject68,
    "InlineObject69": InlineObject69,
    "InlineObject7": InlineObject7,
    "InlineObject8": InlineObject8,
    "InlineObject9": InlineObject9,
    "InlineResponse200": InlineResponse200,
    "InlineResponse2001": InlineResponse2001,
    "InlineResponse20010": InlineResponse20010,
    "InlineResponse20011": InlineResponse20011,
    "InlineResponse2002": InlineResponse2002,
    "InlineResponse2003": InlineResponse2003,
    "InlineResponse2004": InlineResponse2004,
    "InlineResponse2005": InlineResponse2005,
    "InlineResponse2006": InlineResponse2006,
    "InlineResponse2007": InlineResponse2007,
    "InlineResponse2008": InlineResponse2008,
    "InlineResponse2009": InlineResponse2009,
    "InternalServerError": InternalServerError,
    "LedgerEntry": LedgerEntry,
    "LinkBook": LinkBook,
    "LinkBookSection": LinkBookSection,
    "MSMemberHistory": MSMemberHistory,
    "MSMemberHistoryPagination": MSMemberHistoryPagination,
    "MSMemberRedemption": MSMemberRedemption,
    "MSMemberRedemptionPagination": MSMemberRedemptionPagination,
    "MSMembershipTierHistory": MSMembershipTierHistory,
    "MSMembershipTierHistoryPagination": MSMembershipTierHistoryPagination,
    "MSMembershipTierRedemption": MSMembershipTierRedemption,
    "MSMembershipTierRedemptionPagination": MSMembershipTierRedemptionPagination,
    "MSMerchantCreditHistory": MSMerchantCreditHistory,
    "MSMerchantCreditHistoryPagination": MSMerchantCreditHistoryPagination,
    "MSMerchantCreditRedemption": MSMerchantCreditRedemption,
    "MSMerchantCreditRedemptionPagination": MSMerchantCreditRedemptionPagination,
    "MediaFile": MediaFile,
    "Member": Member,
    "MemberSearch": MemberSearch,
    "Merchant": Merchant,
    "MerchantCreditSearch": MerchantCreditSearch,
    "MerchantNotInitialized": MerchantNotInitialized,
    "MerchantURL": MerchantURL,
    "Message": Message,
    "ModuleError": ModuleError,
    "NewsArticle": NewsArticle,
    "OptInList": OptInList,
    "OptInListSource": OptInListSource,
    "OptInListSubscriber": OptInListSubscriber,
    "OutboundSMS": OutboundSMS,
    "PaginatedWTMembers": PaginatedWTMembers,
    "PaginatedWTMerchantCredits": PaginatedWTMerchantCredits,
    "PaginationRequestWithSortOptions": PaginationRequestWithSortOptions,
    "PaymentDesign": PaymentDesign,
    "Performance": Performance,
    "PhoneNumber": PhoneNumber,
    "PickCreateStaticVoucherCampaignWithVoucherExcludeKeyofcreateStaticVoucherCampaignWithVoucherSourceID": PickCreateStaticVoucherCampaignWithVoucherExcludeKeyofcreateStaticVoucherCampaignWithVoucherSourceID,
    "PickMSMemberCreationParamsExcludeKeyofMSMemberCreationParamsMemberIdentifier": PickMSMemberCreationParamsExcludeKeyofMSMemberCreationParamsMemberIdentifier,
    "PickMSMemberExcludeKeyofMSMemberMemberIdentifier": PickMSMemberExcludeKeyofMSMemberMemberIdentifier,
    "PickMSMerchantCreditCreationParamsExcludeKeyofMSMerchantCreditCreationParamsMemberIdentifier": PickMSMerchantCreditCreationParamsExcludeKeyofMSMerchantCreditCreationParamsMemberIdentifier,
    "PickMSMerchantCreditExcludeKeyofMSMerchantCreditMemberIdentifier": PickMSMerchantCreditExcludeKeyofMSMerchantCreditMemberIdentifier,
    "PickPaginationRequestWithoutSortOptionsExcludeKeyofPaginationRequestWithoutSortOptionsIsArchiveIncluded": PickPaginationRequestWithoutSortOptionsExcludeKeyofPaginationRequestWithoutSortOptionsIsArchiveIncluded,
    "PickSSImportedListUpdateParamsExcludeKeyofSSImportedListUpdateParamsId": PickSSImportedListUpdateParamsExcludeKeyofSSImportedListUpdateParamsId,
    "PickSSMobileNumberUpdateParamsExcludeKeyofSSMobileNumberUpdateParamsId": PickSSMobileNumberUpdateParamsExcludeKeyofSSMobileNumberUpdateParamsId,
    "PickSSOptInListCreateParamsExcludeKeyofSSOptInListCreateParamsEmployeeID": PickSSOptInListCreateParamsExcludeKeyofSSOptInListCreateParamsEmployeeID,
    "PickSSOptInListUpdateParamsExcludeKeyofSSOptInListUpdateParamsId": PickSSOptInListUpdateParamsExcludeKeyofSSOptInListUpdateParamsId,
    "PickSSOptInSourceUpdateParamsExcludeKeyofSSOptInSourceUpdateParamsId": PickSSOptInSourceUpdateParamsExcludeKeyofSSOptInSourceUpdateParamsId,
    "PickSSOutboundMessageLogExcludeKeyofSSOutboundMessageLogToCellPhone": PickSSOutboundMessageLogExcludeKeyofSSOutboundMessageLogToCellPhone,
    "PickVSAdvertisementCreditScanExcludeKeyofVSAdvertisementCreditScanRedeemedAtOrRefundedAt": PickVSAdvertisementCreditScanExcludeKeyofVSAdvertisementCreditScanRedeemedAtOrRefundedAt,
    "PickVSAdvertisementCreditUpdateParamsExcludeKeyofVSAdvertisementCreditUpdateParamsId": PickVSAdvertisementCreditUpdateParamsExcludeKeyofVSAdvertisementCreditUpdateParamsId,
    "PickVSCampaignUpdateParamsExcludeKeyofVSCampaignUpdateParamsStartDateOrExpirationDate": PickVSCampaignUpdateParamsExcludeKeyofVSCampaignUpdateParamsStartDateOrExpirationDate,
    "PickVSDynamicVoucherExcludeKeyofVSDynamicVoucherTemporalDecreaseFrequencyType": PickVSDynamicVoucherExcludeKeyofVSDynamicVoucherTemporalDecreaseFrequencyType,
    "PickVSPaymentDesignCreateParamsExcludeKeyofVSPaymentDesignCreateParamsBackgroundImageURLOrCompanyLogoURLOrEmployeeID": PickVSPaymentDesignCreateParamsExcludeKeyofVSPaymentDesignCreateParamsBackgroundImageURLOrCompanyLogoURLOrEmployeeID,
    "PickVSPaymentDesignUpdateParamsExcludeKeyofVSPaymentDesignUpdateParamsId": PickVSPaymentDesignUpdateParamsExcludeKeyofVSPaymentDesignUpdateParamsId,
    "PickVSStaticVoucherExcludeKeyofVSStaticVoucherRedeemedAtOrRefundedAtOrLastViewedAt": PickVSStaticVoucherExcludeKeyofVSStaticVoucherRedeemedAtOrRefundedAtOrLastViewedAt,
    "PickWTAdvertisementCreditCreateParamsExcludeKeyofWTAdvertisementCreditCreateParamsEmployeeID": PickWTAdvertisementCreditCreateParamsExcludeKeyofWTAdvertisementCreditCreateParamsEmployeeID,
    "PickWTEmployeeCreateExcludeKeyofWTEmployeeCreateEmail": PickWTEmployeeCreateExcludeKeyofWTEmployeeCreateEmail,
    "PickWTMemberMemberIDOrFirstNameOrLastNameOrMembershipTierIDOrPointsAccruedOrMobileNumber": PickWTMemberMemberIDOrFirstNameOrLastNameOrMembershipTierIDOrPointsAccruedOrMobileNumber,
    "PickWTMerchantCreditMemberIDOrCreditAmountOrMobileNumber": PickWTMerchantCreditMemberIDOrCreditAmountOrMobileNumber,
    "PickWTMerchantURLUpdateExcludeKeyofWTMerchantURLUpdateUrlID": PickWTMerchantURLUpdateExcludeKeyofWTMerchantURLUpdateUrlID,
    "PickWTNewsArticleUpdateParamsExcludeKeyofWTNewsArticleUpdateParamsId": PickWTNewsArticleUpdateParamsExcludeKeyofWTNewsArticleUpdateParamsId,
    "PickWTPaymentObjectBroadcastExcludeKeyofWTPaymentObjectBroadcastListTypeOrListID": PickWTPaymentObjectBroadcastExcludeKeyofWTPaymentObjectBroadcastListTypeOrListID,
    "PickWTStaticVoucherCampaignCreateParamsExcludeKeyofWTStaticVoucherCampaignCreateParamsStartDateTimeOrExpirationDateTime": PickWTStaticVoucherCampaignCreateParamsExcludeKeyofWTStaticVoucherCampaignCreateParamsStartDateTimeOrExpirationDateTime,
    "PickWTStaticVoucherCreateParamsExcludeKeyofWTStaticVoucherCreateParamsCampaignID": PickWTStaticVoucherCreateParamsExcludeKeyofWTStaticVoucherCreateParamsCampaignID,
    "Plan": Plan,
    "PresignedPost": PresignedPost,
    "PresignedPostFields": PresignedPostFields,
    "Product": Product,
    "PromoCode": PromoCode,
    "ReachPerformanceStats": ReachPerformanceStats,
    "Request": Request,
    "Response": Response,
    "Role": Role,
    "RoleAuditLog": RoleAuditLog,
    "SSOptInSource": SSOptInSource,
    "SimpleSMSBroadcast": SimpleSMSBroadcast,
    "StaticVoucher": StaticVoucher,
    "StaticVoucherCampaign": StaticVoucherCampaign,
    "StaticVoucherCampaignBroadcast": StaticVoucherCampaignBroadcast,
    "StaticVoucherCampaignGroup": StaticVoucherCampaignGroup,
    "StaticVoucherCampaignUpdate": StaticVoucherCampaignUpdate,
    "Status": Status,
    "SubscriptionFeature": SubscriptionFeature,
    "SubscriptionPlan": SubscriptionPlan,
    "SubscriptionProduct": SubscriptionProduct,
    "Tcpa": Tcpa,
    "UpdateStaticVoucherCampaignWithVoucher": UpdateStaticVoucherCampaignWithVoucher,
    "VSCampaignGeneratedMessage": VSCampaignGeneratedMessage,
    "VSCampaignGeneratedMessagePagination": VSCampaignGeneratedMessagePagination,
    "VSDynamicVoucherStatus": VSDynamicVoucherStatus,
    "WTAdvertisementCredit": WTAdvertisementCredit,
    "WTAdvertisementCreditScan": WTAdvertisementCreditScan,
    "WTAdvertisementCreditUpdateParams": WTAdvertisementCreditUpdateParams,
    "WTDynamicVoucher": WTDynamicVoucher,
    "WTDynamicVoucherCreateParams": WTDynamicVoucherCreateParams,
    "WTDynamicVoucherRedemption": WTDynamicVoucherRedemption,
    "WTDynamicVoucherSummary": WTDynamicVoucherSummary,
    "WTDynamicVoucherUpdateParams": WTDynamicVoucherUpdateParams,
    "WTEmployee": WTEmployee,
    "WTEmployeeCreate": WTEmployeeCreate,
    "WTEmployeeUpdate": WTEmployeeUpdate,
    "WTImportedList": WTImportedList,
    "WTLinkBookCreateParams": WTLinkBookCreateParams,
    "WTLinkBookSectionCreateParams": WTLinkBookSectionCreateParams,
    "WTLinkBookSectionUpdateParams": WTLinkBookSectionUpdateParams,
    "WTLinkBookUpdateParams": WTLinkBookUpdateParams,
    "WTMember": WTMember,
    "WTMemberCreationParams": WTMemberCreationParams,
    "WTMembershipTier": WTMembershipTier,
    "WTMembershipTierCreationParams": WTMembershipTierCreationParams,
    "WTMembershipTierUpdateParams": WTMembershipTierUpdateParams,
    "WTMembershipTierWithMemberCount": WTMembershipTierWithMemberCount,
    "WTMerchantCredit": WTMerchantCredit,
    "WTMerchantCreditCreationParams": WTMerchantCreditCreationParams,
    "WTMerchantURLCreate": WTMerchantURLCreate,
    "WTMerchantURLUpdate": WTMerchantURLUpdate,
    "WTNewsArticleCreateParams": WTNewsArticleCreateParams,
    "WTNewsArticleUpdateParams": WTNewsArticleUpdateParams,
    "WTOptInList": WTOptInList,
    "WTOptInListCreationParams": WTOptInListCreationParams,
    "WTPaymentDesign": WTPaymentDesign,
    "WTPaymentDesignCreateParams": WTPaymentDesignCreateParams,
    "WTPaymentDesignUpdateParams": WTPaymentDesignUpdateParams,
    "WTPerformanceCreateParams": WTPerformanceCreateParams,
    "WTPerformanceUpdateParams": WTPerformanceUpdateParams,
    "WTPosMachine": WTPosMachine,
    "WTPosMachineCreateParams": WTPosMachineCreateParams,
    "WTPosMachineUpdateParams": WTPosMachineUpdateParams,
    "WTPromoCodeCreateParams": WTPromoCodeCreateParams,
    "WTPromoCodeUpdateParams": WTPromoCodeUpdateParams,
    "WTRole": WTRole,
    "WTStaticVoucher": WTStaticVoucher,
    "WTStaticVoucherCampaign": WTStaticVoucherCampaign,
    "WTStaticVoucherCreateParams": WTStaticVoucherCreateParams,
    "WTStaticVoucherUpdateParams": WTStaticVoucherUpdateParams,
    "WTWalletPageView": WTWalletPageView,
    "WTWalletPageViewGeoPoint": WTWalletPageViewGeoPoint,
    "WalletPageView": WalletPageView,
    "Webpage": Webpage,
}

export class ObjectSerializer {
    public static findCorrectType(data: any, expectedType: string) {
        if (data == undefined) {
            return expectedType;
        } else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        } else if (expectedType === "Date") {
            return expectedType;
        } else {
            if (enumsMap[expectedType]) {
                return expectedType;
            }

            if (!typeMap[expectedType]) {
                return expectedType; // w/e we don't know the type
            }

            // Check the discriminator
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType; // the type does not have a discriminator. use it.
            } else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if(typeMap[discriminatorType]){
                        return discriminatorType; // use the type given in the discriminator
                    } else {
                        return expectedType; // discriminator did not map to a type
                    }
                } else {
                    return expectedType; // discriminator was not present (or an empty string)
                }
            }
        }
    }

    public static serialize(data: any, type: string) {
        if (data == undefined) {
            return data;
        } else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        } else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType: string = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData: any[] = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.serialize(datum, subType));
            }
            return transformedData;
        } else if (type === "Date") {
            return data.toISOString();
        } else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) { // in case we dont know the type
                return data;
            }

            // Get the actual type of this object
            type = this.findCorrectType(data, type);

            // get the map for the correct type.
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance: {[index: string]: any} = {};
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
            }
            return instance;
        }
    }

    public static deserialize(data: any, type: string) {
        // polymorphism may change the actual type.
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        } else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        } else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType: string = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData: any[] = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.deserialize(datum, subType));
            }
            return transformedData;
        } else if (type === "Date") {
            return new Date(data);
        } else {
            if (enumsMap[type]) {// is Enum
                return data;
            }

            if (!typeMap[type]) { // dont know the type
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.name] = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type);
            }
            return instance;
        }
    }
}

export interface Authentication {
    /**
    * Apply authentication settings to header and query params.
    */
    applyToRequest(requestOptions: localVarRequest.Options): Promise<void> | void;
}

export class HttpBasicAuth implements Authentication {
    public username: string = '';
    public password: string = '';

    applyToRequest(requestOptions: localVarRequest.Options): void {
        requestOptions.auth = {
            username: this.username, password: this.password
        }
    }
}

export class HttpBearerAuth implements Authentication {
    public accessToken: string | (() => string) = '';

    applyToRequest(requestOptions: localVarRequest.Options): void {
        if (requestOptions && requestOptions.headers) {
            const accessToken = typeof this.accessToken === 'function'
                            ? this.accessToken()
                            : this.accessToken;
            requestOptions.headers["Authorization"] = "Bearer " + accessToken;
        }
    }
}

export class ApiKeyAuth implements Authentication {
    public apiKey: string = '';

    constructor(private location: string, private paramName: string) {
    }

    applyToRequest(requestOptions: localVarRequest.Options): void {
        if (this.location == "query") {
            (<any>requestOptions.qs)[this.paramName] = this.apiKey;
        } else if (this.location == "header" && requestOptions && requestOptions.headers) {
            requestOptions.headers[this.paramName] = this.apiKey;
        } else if (this.location == 'cookie' && requestOptions && requestOptions.headers) {
            if (requestOptions.headers['Cookie']) {
                requestOptions.headers['Cookie'] += '; ' + this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
            else {
                requestOptions.headers['Cookie'] = this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
        }
    }
}

export class OAuth implements Authentication {
    public accessToken: string = '';

    applyToRequest(requestOptions: localVarRequest.Options): void {
        if (requestOptions && requestOptions.headers) {
            requestOptions.headers["Authorization"] = "Bearer " + this.accessToken;
        }
    }
}

export class VoidAuth implements Authentication {
    public username: string = '';
    public password: string = '';

    applyToRequest(_: localVarRequest.Options): void {
        // Do nothing
    }
}

export type Interceptor = (requestOptions: localVarRequest.Options) => (Promise<void> | void);
