// wallet/model/reachPerformanceStats.ts
/**
 * wallet-api
 * API
 *
 * The version of the OpenAPI document: 1.1.6
 * Contact: development@wallet.inc
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { RequestFile } from './models';

export class ReachPerformanceStats {
    /**
    * Denotes the number of SMSes sent
    */
    'sentCount': number;
    /**
    * Denotes the number of SMSes delivered
    */
    'deliveredCount': number;
    /**
    * Denotes the count of SMSes viewed (by clicking on the link sent as part of the SMS)
    */
    'viewedCount': number;
    /**
    * Denotes the count of redemptions
    */
    'redemptionsCount': number;
    /**
    * Denotes the count of refunds
    */
    'refundsCount': number;

    static discriminator: string | undefined = undefined;

    static attributeTypeMap: Array<{name: string, baseName: string, type: string}> = [
        {
            "name": "sentCount",
            "baseName": "sentCount",
            "type": "number"
        },
        {
            "name": "deliveredCount",
            "baseName": "deliveredCount",
            "type": "number"
        },
        {
            "name": "viewedCount",
            "baseName": "viewedCount",
            "type": "number"
        },
        {
            "name": "redemptionsCount",
            "baseName": "redemptionsCount",
            "type": "number"
        },
        {
            "name": "refundsCount",
            "baseName": "refundsCount",
            "type": "number"
        }    ];

    static getAttributeTypeMap() {
        return ReachPerformanceStats.attributeTypeMap;
    }
}
